### PR TITLE
[lldb] Use unique line count in swift_task_switch logic

### PR DESCRIPTION
**Pull request by kastiglione**
When stepping into an async function, the compiler can emit two cases for the first async function split:
1. A trampoline that immediately performs async dispatch via `swift_task_switch`
2. The initial block of the async function's body

In the second case, step-in is done. In the second case, lldb does additional work to stop when the runtime dispatches to the async function (including setting a conditional internal breakpoint, and more).

lldb needs to distinguish between these two cases, in order to know which stepping logic to perform.

In the first case, the trampoline would ideally have no associated line table entries for the trampoline. However that's not been the case. Instead there has been a single (non-zero) line table entry for the trampoline. The logic lldb uses is to check the line table, and if there's only one line table entry, conclude that it's a trampoline. But, a recent update to the swift compiler has resulted in two line table entries for the trampoline, but both are for the same line – in other words there's still a single line referenced.

This changes the logic to use a set to track the line table entries. If only one unique line is found, conclude the function is a trampoline.

Ultimately, making swiftc emit no line table for these trampolines is the best fix.

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
Working on a test.

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test macOS

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@adrian-prantl this is ready

File: lldb/test/API/lang/swift/async/stepping/step-in/task-switch/TestSwiftTaskSwitch.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @skipIf(oslist=["windows", "linux"])
+    def test(self):
+        """Test conditions for async step-in."""
+        self.build()
+
+        src = lldb.SBFileSpec("main.swift")
+        target, _, thread, _ = lldbutil.run_to_source_breakpoint(self, "await f()", src)
+        self.assertEqual(thread.frame[0].function.mangled, "$s1a5entryO4mainyyYaFZ")
+
+        function = target.FindFunctions("$s1a5entryO4mainyyYaFZTQ0_")[0].function
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/async/stepping/step-in/task-switch/TestSwiftTaskSwitch.py:18]: can you assert that function is non-null? Will be easier to debug that way.

Comment (kastiglione) [reply lldb/test/API/lang/swift/async/stepping/step-in/task-switch/TestSwiftTaskSwitch.py:18]: https://github.com/apple/llvm-project/pull/6908

**Pull request merged**